src/sentry_sampling_context.h

```diff
@@ -0,0 +1,16 @@
+// sentry_sampling_context.h
+#ifndef SENTRY_SAMPLING_CONTEXT_H_INCLUDED
+#define SENTRY_SAMPLING_CONTEXT_H_INCLUDED
+
+#include "sentry_tracing.h"
+
+typedef struct sentry_sampling_context_s {
+    sentry_transaction_context_t *transaction_context;
+    void *custom_sampling_context; // TODO what type should this be?
```

Comment:
Do we want to provide something similar to [the cocoa SDK](https://github.com/getsentry/sentry-cocoa/blob/main/Sources/Sentry/SentrySamplingContext.m#L16), or is it up to the users to define what type and how they'll use it in the tracesSampler callback?

e.g.
```c
typedef struct {
    char *key;
    void *value;
} custom_sampling_context_entry_t;

typedef struct {
    custom_sampling_context_entry_t *entries;
    size_t entry_count;
} custom_sampling_context_t;
```